### DISCUSSION TITLE
AzureDevOps Private registry returns 401

### DISCUSSION THREAD

**Discussion by mkercheval-tva** [vercel/vercel#10229] [General]
### Description

https://vercel.com/guides/using-private-dependencies-with-vercel

We are building our app in a Azure DevOps pipeline and have a few packages in a private Azure Artifacts feed. We added a the environment variable, but we are still getting a 401. Is there a good example for Azure Devops?

Below is an example of our environment variable 

//pkgs.dev.azure.com/xxxxxxxxxxxxxxxxx/_packaging/xxxxxxxxxx/npm/registry/:_authToken=0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000=

**Comment by styles102** [comment]
I was also struggling with this, and found the following solution. Below is an example of the .npmrc file that's been added to vercel.

`
registry=https://registry.npmjs.org/
@SOURCE:registry=https://pkgs.dev.azure.com/{Organisation}/_packaging/{PackageName}/npm/registry/
//pkgs.dev.azure.com/{Organisation}/_packaging/{PackageName}/npm/registry/:username={AZURE_USERNAME}
//pkgs.dev.azure.com/{Organisation}/_packaging/{PackageName}/npm/registry/:_password={AZURE_PAT_BASE_64}
always-auth=true
`

I found that you need to add the additional two lines to provide the username and password. The password is the azure personal access token but base64'd.

Hope this helps!
